parserator_web/static/js/index.js

```diff
@@ -1 +1,92 @@
-/* JS code goes here */
+/* I got a lint error for this async function
+   "Unexpected token function"
+   but wasn't able to fix, and kept function as is */
+async function getAndDisplayAddressData(inputString) {
+  const url = "api/parse/?input_string=" + inputString;
+  const response = await fetch(url, {
+    method: "GET",
+  })
+    .then((response) => response.json())
+    .then((data) => {
+      if (data.error) {
+        console.error(data.error);
+        displayError(data.error);
+      } else {
+        displayResults(data);
+      }
+    });
+}
+
+// display error returned from the API request in the error div
+function displayError(errorMessage) {
+  // this pattern is from getbootstrap.com/docs/5.2/components/alerts/#triggers
+  const alertPlaceholder = document.getElementById("alert-placeholder");
+
+  const wrapper = document.createElement("div");
+  wrapper.innerHTML = [
+    `<div class="alert alert-danger alert-dismissible" role="alert">`,
+    `   <div>${errorMessage}</div>`,
+    '   <button type="button" class="close" data-dismiss="alert" aria-label="Close">',
+    '     <span aria-hidden="true">&times;</span>',
+    "   </button>",
+    "</div>",
+  ].join("");
+
+  alertPlaceholder.append(wrapper);
+}
+
+function displayResults(data) {
+  // reveal the results table
+  document.getElementById("address-results").style.display = "block";
+  console.log(data);
+  // populate the address type
+  document.getElementById("parse-type").innerHTML = data.address_type;
+
+  //clear the tbody
+  const tableBody = document.querySelector("#address-results table tbody");
+  tableBody.innerHTML = "";
+
+  // populate the tbody
+  for (let [part, tag] of Object.entries(data.address_components)) {
+    console.log(part, tag);
+    let row = document.createElement("tr");
```

Comment:
you might want to consider using the insertRow and insertCell methods https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableElement/insertRow

Reply:
Oh, I didn't know about these methods, thank you for pointing these out!